features/admin.users.v1/components/user-profile.tsx

```diff
@@ -652,6 +672,131 @@ export const UserProfile: FunctionComponent<UserProfilePropsInterface> = (
         setPrimaryValues(tempPrimaryValues);
     };
 
+    useEffect(() => {
+        if (!isAuthenticatedUserFetchRequestLoading) {
+            if (authenticatedUserProfileInfoData) {
+                setAuthenticatedUserRoles(authenticatedUserProfileInfoData?.roles);
+            }
+        }
+    }, [ isAuthenticatedUserFetchRequestLoading ]);
+
+    useEffect(() => {
+        if (impersonation_artifacts != undefined && impersonation_artifacts != null) {
+
+            const id_token: any = new URLSearchParams(impersonation_artifacts)
+                .get(UserManagementConstants.ID_TOKEN);
+            const subject_token: any = new URLSearchParams(impersonation_artifacts)
+                .get(UserManagementConstants.SUBJECT_TOKEN);
+
+            if (id_token && subject_token) {
+                setIdToken(id_token);
+                setSubjectToken(subject_token);
+            }
+        }
+    }, [ impersonation_artifacts ]);
+
+    const handleMessage = (event: any) => {
+        if (event.data === "impersonation-authorize-request-complete") {
+            impersonation_artifacts = sessionStorage.getItem("impersonation_artifacts").substring(1);
+            const id_token: any = new URLSearchParams(impersonation_artifacts).get("id_token");
+            const subject_token: any = new URLSearchParams(impersonation_artifacts).get("subject_token");
+
+            if (id_token && subject_token) {
+                setIdToken(id_token);
+                setSubjectToken(subject_token);
+            }
+        }
+    };
+
+    useEffect(() => {
+        window.addEventListener("message", handleMessage);
+
+        return () => window.removeEventListener("message", handleMessage);
+    }, []);
+
+    useEffect(() => {
+
+        if (idToken && subjectToken) {
+
+            const formData: any = {
+                actor_token: idToken,
+                actor_token_type: "urn:ietf:params:oauth:token-type:id_token",
+                client_id: "Pfao0gjJ07be0gf29PzEcLFKmUIa",
+                code_verifier: codeVerifier,
+                grant_type: "urn:ietf:params:oauth:grant-type:token-exchange",
+                requested_token_type: "urn:ietf:params:oauth:token-type:access_token",
+                subject_token: subjectToken,
+                subject_token_type: "urn:ietf:params:oauth:token-type:jwt"
+            };
+
+            const urlEncodedData: any = new URLSearchParams(formData).toString();
+
+            const requestConfig: any = {
+                attachToken: false,
+                data: urlEncodedData,
+                headers: {
+                    "Content-Type": "application/x-www-form-urlencoded"
+                },
+                method: "post",
+                shouldAttachIDPAccessToken: false,
```

Comment:
This wouldn't be necessary.

Reply:
Fixed with 82d4ff507d55ec7ed6c0b57080b6e89963594857